### PR TITLE
[cxx-interop] Allow specific getters/setters to be imported as comput…

### PR DESCRIPTION
**Pull request by zoecarver**
…ed properties.

Adds `SWIFT_COMPUTED_PROPERTY`. Refs `-cxx-interop-getters-setters-as-properties`.

### CODE REVIEW

**Comment by zoecarver** [comment]
rdar://107757246

**Comment by zoecarver** [comment]
@swift-ci please test

File: test/Interop/Cxx/ergonomics/explicit-computed-properties.swift [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    requires cplusplus
+}
+
+//--- Inputs/test.h
+
+#define SWIFT_COMPUTED_PROPERTY \
+  __attribute__((swift_attr("import_computed_property")))
+
+struct Record {
+  int getX() SWIFT_COMPUTED_PROPERTY { return 42; }
```
Comment (hyp) [inline test/Interop/Cxx/ergonomics/explicit-computed-properties.swift:17]: Can you test this also applying to both getter and setter?

**Comment by ravikandhadai** [comment]
@zoecarver wondering how will the errors look if the attribute is used on a method with arguments? A test for that? Also, wondering if there can be name collisions when importing the two different methods using this attributes. If so, how the errors would look?

**Comment by zoecarver** [comment]
We can add further testing after the fact on main.

**Comment by zoecarver** [comment]
Want to get this into 5.9 asap

**Review by ravikandhadai — COMMENTED** [review]
LGTM

**Comment by ravikandhadai** [comment]
> @zoecarver wondering how will the errors look if the attribute is used on a method with arguments? A test for that? Also, wondering if there can be name collisions when importing the two different methods using this attributes. If so, how the errors would look?

IIUC if the attribute is wrongly used, it wouldn't have any effect.

**Pull request merged**